wp-lazy-loading.php

```diff
@@ -32,10 +32,8 @@
 function _wp_lazy_loading_initialize_filters() {
 	// The following filters would be merged into core.
 	foreach ( array( 'the_content', 'the_excerpt', 'comment_text', 'widget_text_content' ) as $filter ) {
-		// Before parsing blocks and shortcodes.
-		// TODO: Comments do not support images. Revisit.
-		// TODO: This should not exclude images from dynamic blocks and shortcodes. Look at fixing the filter priority.
-		add_filter( $filter, 'wp_add_lazy_load_attributes', 8 );
+		// After parsing blocks and shortcodes.
+		add_filter( $filter, 'wp_add_lazy_load_attributes', 25 );
```

Comment:
Can you give your thinking on why you chose 25 as priority specifically?

Looping in @spacedmonkey because he originally set it to 8.

Reply:
25 is somewhat arbitrary for "after all other default filters".

Not sure why it was set to 8, but that means we were not adding the attribute to images added with shortcodes, like to the (old style) galleries, etc. Running it "early" means it misses few cases where images may be added, don't see why these images should be skipped.

Reply:
8 was set because this filter should be applied before blocks are rendered.

Reply:
> ...should be applied before blocks are rendered

What's the reason to exclude dynamic blocks? Did you run into some troubles with any of them?

Generally thinking we want the attribute on all images in all cases. Then, if there are cases where lazy-loading doesn't work well, find a way to skip adding it if possible.

Reply:
Not applying these filters early is the desired behaviour here. Adding this attribute to rendered blocks, shortcodes and other filtered content as we have no idea what effect that will have on third party code. I think they would be breakages and backlash if we opt everyone in. Take for example a tracking pixel loaded in via a shortcode or block. If this is then lazy loaded, then it would be a breakage. 

We should make sure all core functionality like galleries shortcodes supports lazy loading attribute. We should also have a way for third party code to opt in.

Reply:
> ...we have no idea what effect that will have on third party code.

This is the main purpose of this plugin, to test such cases. Frankly I don't see much difference between HTML content added by the users directly (may be pasted, etc.) and HTML content added by plugins. Excluding images that may be added by the user using a plugin that adds a shortcode or a specific block is not a good idea imho, at least for now. If any breakage is detected during testing, lets look at the specific cases.

(Having said that, we are running late... Looks like this feature will miss WP 5.4 if this plugin is not released asap, as there will not be sufficient time to test).

> Take for example a tracking pixel loaded in via a shortcode or block.

This case is specifically discussed in the HTML specs. Also, as the browser downloads the first 2KB of each image, tracking pixels will work regardless of the attribute or its value (these images are usually smaller than 2KB anyway).